modules/c/nitf/source/ImageIO.c

```diff
@@ -5855,12 +5855,16 @@ NITFPRIV(void) nitf_ImageIOControl_destruct(_nitf_ImageIOControl ** cntl)
          * They are allocated
          */
         nBlockCols = cntlActual->nBlockIO / cntlActual->numBandSubset;
-        blocks = &(cntlActual->blockIO[0][0]);
+
         for (i = 0; i < nBlockCols; ++i)
         {
-            if (blocks[i].blockControl.freeFlag)
+            for (j = 0; j < cntlActual->numBandSubset; ++j)
             {
-                NITF_FREE(blocks[i].blockControl.block);
+                blocks = &(cntlActual->blockIO[i][j]);
```

Comment:
So basically this previously leaked for multi-band NITFs?

Reply:
Yes